$MirOS: ports/graphics/jpeg/patches/patch-wrjpgcom_c,v 1.1 2007/01/15 22:01:35 tg Exp $
$OpenBSD: patch-wrjpgcom_c,v 1.1 2009/10/14 13:03:44 naddy Exp $

	Discovered by Felix von Leitner, thanks. Fix by tg@
	More fix by naddy@

--- wrjpgcom.c.orig	Thu Oct 23 04:46:42 1997
+++ wrjpgcom.c	Mon Jan 15 21:59:57 2007
@@ -453,7 +453,7 @@ main (int argc, char **argv)
 	comment_arg = (char *) malloc((size_t) MAX_COM_LENGTH);
 	if (comment_arg == NULL)
 	  ERREXIT("Insufficient memory");
-	strcpy(comment_arg, argv[argn]+1);
+	strlcpy(comment_arg, argv[argn]+1, (size_t) MAX_COM_LENGTH);
 	for (;;) {
 	  comment_length = (unsigned int) strlen(comment_arg);
 	  if (comment_length > 0 && comment_arg[comment_length-1] == '"') {
@@ -462,8 +462,8 @@ main (int argc, char **argv)
 	  }
 	  if (++argn >= argc)
 	    ERREXIT("Missing ending quote mark");
-	  strcat(comment_arg, " ");
-	  strcat(comment_arg, argv[argn]);
+	  strlcat(comment_arg, " ", MAX_COM_LENGTH);
+	  strlcat(comment_arg, argv[argn], MAX_COM_LENGTH);
 	}
       }
       comment_length = (unsigned int) strlen(comment_arg);
